### PR TITLE
[reconfiguration] Replace epoch store with committee store

### PR DESCRIPTION
**Pull request by lxfind**
Built on top of https://github.com/MystenLabs/sui/pull/4923, now that we commit the committee to the last checkpoint of each epoch, we no longer need a separate epoch data structure.
This PR replaces it with a side index table that stores the epoch to committee map.

### CODE REVIEW

**Review by mystenmark — COMMENTED** [review]
My big question (which I mentioned in slack) is:

Previously, the Epoch datastructures for a hash-chain from one epoch to the next. We still have the same committments after this PR, but you have to examine every checkpoint in the epoch in order to verify the hash chain back to the previous epoch.

I'm not sure how much that matters but it would be good if someone could come up with a convincing argument for why we don't (or do) care about this.

**Comment by lxfind** [comment]
A commit to the previous "last checkpoint of the epoch" should be easy to add though. I will add a TODO there before merging

File: crates/sui-types/src/messages.rs [hunk 1]
```diff
@@ -2081,214 +2081,3 @@ pub enum QuorumDriverResponse {
...
-
-pub type EpochInfoDigest = [u8; 32];
-
-#[derive(Debug, Clone, Serialize, Deserialize)]
-pub struct EpochEnvelop<S> {
-    pub epoch_info: EpochInfo,
-    pub auth_sign_info: S,
-}
-
-pub type GenesisEpoch = EpochEnvelop<EmptySignInfo>;
```
Comment (longbowlu) [inline crates/sui-types/src/messages.rs:2143]: it's removed but not sure if worth asking: don't have we validators sigs in in genesis.blob to construct a certified `GenesisEpoch`?

File: crates/sui-core/src/epoch/committee_store.rs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+fn committee_table_default_config() -> Options {
+    default_db_options(None, None).1
+}
+
+impl CommitteeStore {
+    pub fn new(path: PathBuf, genesis_committee: &Committee, db_options: Option<Options>) -> Self {
+        let committee_store = Self::open_tables_read_write(path, db_options, None);
+        if committee_store.database_is_empty() {
+            committee_store
+                .init_genesis_epoch(genesis_committee.clone())
```
Comment (longbowlu) [inline crates/sui-core/src/epoch/committee_store.rs:36]: is genesis_epoch or a similar concept stored on chain today? e.g. when one reverse traverse the hashchain , how do they verify the genesis committee/state? 

There's certified genesis.blob but is it stored on chain?

**Review by longbowlu — APPROVED** [review]

File: crates/sui-types/src/messages.rs (hunk 1 above)
Comment (lxfind) [reply crates/sui-types/src/messages.rs:2143]: The GenesisEpoch today is not certified. But that's a good question. I am not quite sure what we plan to do with the genesis committee

File: crates/sui-core/src/epoch/committee_store.rs (hunk 2 above)
Comment (lxfind) [reply crates/sui-core/src/epoch/committee_store.rs:36]: I think the assumption is that the genesis committee is the one thing that everyone has to trust through some offline methods (e.g. download from a link)

**Pull request merged**